internal/controllers/user_integrations_controller.go

```diff
@@ -1933,6 +1953,49 @@ func (udc *UserDevicesController) registerDeviceTesla(c *fiber.Ctx, logger *zero
 	return c.SendStatus(fiber.StatusNoContent)
 }
 
+func (udc *UserDevicesController) getTeslaAuthFromCache(ctx context.Context, userID string) (*RegisterDeviceIntegrationRequest, error) {
+	user, err := udc.usersClient.GetUser(ctx, &pb.GetUserRequest{Id: userID})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fetch user information")
```

Comment:
We should stop using https://github.com/pkg/errors for new code. This package has been archived for a couple of years since error wrapping was added to the standard library.
 ```suggestion
		return nil, fmt.Errorf("could not fetch user information: %w", err)
```